loaders/authentication.js

```diff
@@ -0,0 +1,38 @@
+const fetch = require('node-fetch');
+const dotenv = require('dotenv').config();
+
+//base URL of cigarra authentication
+const URL = 'https://sigarra.up.pt/feup/pt/mob_val_geral.autentica?'
+
+//build URL with credentials
+let urlRequest = URL + 'pv_login=' + process.env.UP_USER + '&pv_password=' + process.env.UP_PWD;
+
+let cookie = null;
+
+//Called sigarra api to make authentication
+async function updateCookie(){
+   try {
+        const response = await fetch(urlRequest, {
+            method: 'POST'
+        });
+        if(response.ok){
+            return await response.json();
+        }
+        throw new Error('Request Failed: ' + response.status + "-" + response.statusText);
+
+   }catch (error){
+       console.log(error);
+   }
+}
+
+//give value to cookie
+async function getCookie(){
+    if(cookie == null){
+        cookie = await updateCookie();
```

Comment:
It's not extremely helpful to store the whole response. We can just store the cookies in the response headers. From my tests, the relevant variables were `SI_SESSION` and `SI_SECURITY`.

To get this you can do something like
```js
const res = await fetch(params);
const raw_cookie = res.headers.get('set-cookie');
```

The cookies will be something like: `SI_SESSION=numbers;path=/;HttpOnly, SI_SECURITY=redacted;path=/;secure;HttpOnly`, which need to be parsed simply into `SI_SESSION=numbers;SI_SECURITY=redacted`.  
I think simply taking the beginning of the string until each semicolon would work, but [check this link](https://stackoverflow.com/questions/34815845/how-to-send-cookies-with-node-fetch) for further details.

Reply:
Ok. That was something that we had questions about. The value of the cookie that we where getting doesn't felt right.

Reply:
And to add to what @Andrepereira2001 said, we had some doubts about where we would be able to get the time that our authentication will last. We tried to organize our code to be able to check if the cookie that we have is valid, otherwise we should make the authentication request...

Reply:
I'm not sure that the server responds with the cookie's expiration period (I don't think so).

As such we need to try and figure out its expiration time (using a script or something of the sort) and then add that logic here.

We could have a variable called `COOKIE_TTL_MINUTES` with the respective value. Then, when fetching the cookie, other than storing its value in a variable, we could also store the time after which it will be invalid. Getting the current date using `new Date` and then adding the cookie TTL in minutes to it would do the trick.

Granted, this does not consider that the cookie's lifespan may be extended by it being used in requests. However, we can use this naïve implementation for now and see how things will go in the future.

We can talk about this in further detail when we meet up if this was not clear.